app/controllers/Checkout.scala

```diff
@@ -283,22 +283,18 @@ object Checkout extends Controller with LazyLogging with CatalogProvider {
       }
   }
 
-  def thankYou() = NoCacheAction { implicit request =>
+  def thankYou() = NoCacheAction.async { implicit request =>
     implicit val resolution: TouchpointBackend.Resolution = TouchpointBackend.forRequest(PreSigninTestCookie, request.cookies)
     implicit val tpBackend = resolution.backend
     val session = request.session
 
     val sessionInfo = for {
       subsName <- session.get(SubsName)
-      plan <- session.get(RatePlanId).map(ProductRatePlanId).flatMap(p => catalog.allSubs.flatten.find(_.id == p))
-      ratePlanId <- session.get(RatePlanId)
-      currencyStr <- session.get(SessionKeys.Currency)
-      currency <- Currency.fromString(currencyStr)
       startDate <- session.get(StartDate)
-    } yield (subsName, plan, currency, startDate)
+    } yield (subsName, startDate)
```

Comment:
The 'startDate' being in the session helps genericise this page as it gets round the fact that subscriptions of different products base their fulfilment and payment off different date triggers.